stdsimd-test/assert-instr-macro/src/lib.rs

```diff
@@ -70,8 +70,11 @@ pub fn assert_instr(
                 }
             };
         }
-        let attrs = item.attrs
-            .iter()
+
+        let attrs = match item {
+            syn::Item::Fn(ref item) => &item.attrs,
+            _ => panic!("item must be a function!"),
+        }.iter()
```

Comment:
Can this be done like this?

```rust
let attrs = func.attrs.iter()
```
I really don't know a lot about `syn` but it looks like the match with `item` is done previously and you can already access the attrs from `func`.